lib/common/socketpool.c

```diff
@@ -82,36 +84,49 @@ static void destroy_attached(struct pool_entry_t *entry)
     destroy_detached(entry);
 }
 
-static void destroy_expired(h2o_socketpool_t *pool)
+/* caller should lock the mutex */
+static uint64_t destroy_expired_locked(h2o_socketpool_t *pool)
 {
-    if (pool->_interval_cb.loop == NULL)
-        return;
+    if (pool->_interval_cb.loop != NULL) {
+        uint64_t now_ms = h2o_now(pool->_interval_cb.loop);
+        uint64_t expire_before = now_ms - pool->timeout;
+        while (!h2o_linklist_is_empty(&pool->_shared.sockets)) {
+            struct pool_entry_t *entry = H2O_STRUCT_FROM_MEMBER(struct pool_entry_t, all_link, pool->_shared.sockets.next);
+            if (entry->added_at > expire_before) {
+                return entry->added_at + pool->timeout - now_ms;
+            }
+            destroy_attached(entry);
+            __sync_sub_and_fetch(&pool->_shared.count, 1);
+            __sync_sub_and_fetch(&pool->_shared.pooled_count, 1);
+        }
+    }
+    return SIZE_MAX;
```

Comment:
UINT64_MAX?

Reply:
Yes, should be `UINT64_MAX` for 32 bit system.